### DISCUSSION TITLE
Extract optimized 3d geometry for CAD software

### DISCUSSION THREAD

**Discussion by umarsajjad1992** [mdolab/MACH-Aero#164] [Ideas]
I wanted to use the output of my optimization run to add the optimized wing to the fuselage of my aircraft and was wondering how to use the output of the optimization to work/open with a CAD software such as SolidWorks/CATIA?

**Comment by eirikurj** [comment]
Hi, please take a look at the [pyGeo documentation](https://mdolab-pygeo.readthedocs-hosted.com/en/latest/update_pygeo.html), describing how to deform the initial geometry to obtain an optimized geometry.

**Reply by umarsajjad1992** [reply]
That's not what I was asking. After getting an optimized geometry how can I export that into a readable format other than cgns for CAD softwares such as Solidworks?

**Reply by A-CGray** [reply]
The geometry output from your optimisation is a mesh so the only way to get that into CAD would be as a tessellated  geometry like an STL file, which is not very useful in CAD programs. The page Eirikur linked to shows you how to use pyGeo to deform an iges file of your original geometry into an iges file of your optimised geometry, which is much more useful in CAD.
